.github/workflows/build-compatibility.yml

```diff
@@ -9,31 +9,32 @@ on:
 
 jobs:
   build:
-    name: Run code quality checks
-    runs-on: ubuntu-latest
+    name: Node ${{ matrix.node_version }} on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        node-version: [14.x]
+        os:
+          - ubuntu-latest
+        node_version:
+          - 12
+          - 14
+          - 15
     steps:
       - name: Checkout repository
         uses: actions/checkout@v2
       - name: Use Node.js ${{ matrix.node-version }}
         uses: actions/setup-node@v1
         with:
-          node-version: ${{ matrix.node-version }}
+          node-version: ${{ matrix.node_version }}
       - name: Cache dependencies
         uses: actions/cache@v2
         with:
           path: |
             **/node_modules
-          key: ${{ runner.os }}-${{ hashFiles('**/package-lock.json') }}
+          key: ${{ matrix.os }}-${{ matrix.node_version }}-${{ hashFiles('**/package-lock.json') }}
       - name: Install dependencies
-        run: npm ci
-      - name: Check formatting
-        run: npm run check-format
-      - name: Check linting
-        run: npm run check-lint
-      - name: Check Typescript
-        run: npm run check-types
+        run: npm ci --production
       - name: Check App compilation
-        run: npm run build
+        run: npm run build:ci
+        env:
+          CI: false
```

Comment:
This is confusing, we're asking to run the command specific for CI but with CI=false.
I guess the problem arises from the package.json, where we make the distinction:
```json
"build": "CI=false react-scripts build",
"build:ci": "react-scripts build",
```
I think we could keep just one npm command, such as `"build": "react-scripts build"`, and prepend env variables only where needed. So here we would run `CI=false npm run build`